src/api/registration/EventHandlerRegistration.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2018 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { OnEvent } from "@atomist/automation-client/onEvent";
+import { Maker } from "@atomist/automation-client/util/constructionUtils";
+
+/**
+ * Type for registering event handlers.
+ */
+export interface EventHandlerRegistration<EVENT = any, PARAMS = any> {
```

Comment:
Isn't there a super interface we can capture the commonality in? It won't change usage so it doesn't affect merge